spec/generator/target.rb

```diff
@@ -0,0 +1,243 @@
+    #
+    #   Matches either literal `p` or literal `q`.
+    #
+    #   Matching literals may be more than one character in length. More than
+    #   two literals may be specified.
+    #
+    #   Equivalent to pattern alternation in regexp.
+    #
+    #   ---
+    #
+    #   ### Pattern: \
+    #
+    #   Escapes the next meta-character.
+    #
+    #   ---
+    #
+    #   ### Examples
+    #
+    #   Consider these to be evaluated in the source root of
+    #   [JSONKit](https://github.com/johnezang/JSONKit).
+    #
+    #       "JSONKit.?"    #=> ["JSONKit.h", "JSONKit.m"]
+    #       "*.[a-z][a-z]" #=> ["CHANGELOG.md", "README.md"]
+    #       "*.[^m]*"      #=> ["JSONKit.h"]
+    #       "*.{h,m}"      #=> ["JSONKit.h", "JSONKit.m"]
+    #       "*"            #=> ["CHANGELOG.md", "JSONKit.h"]
+
+    #-----------------------------------------------------------------------#
+
+    # @return [Array or String] files to include for the target.
+    #                           Supports regular expressions,
+    #                           Defaults to: ["./<Target Name>/*\*/\*.*"]
+    #
+    # @example
+    #
+    #     spec.include_files = "Classes/**/*.{h,m}"
+    #
+    attr_accessor :include_files
+
+    # @return [Array or String] files to exclude for the target.
+    #                           Supports regular expressions
+    #
+    # @example
+    #
+    #    target.exclude_files = ["Classes/**/unused.{h,m}"]
+    #
+    attr_accessor :exclude_files
+
+    # @!group Frameworks
+
+    # @return [Array<String>] system frameworks to include for the target
+    #                         Defaults to:
+    #                          - ["Foundation", "UIKit"] for iOS
+    #                          - ["Cocoa"] for OSX
+    #
+    # @example
+    #
+    #    target.system_frameworks = ["Foundation"]
+    #
+    attr_accessor :system_frameworks
+
+    # @return [Array<Target>] targets to use as dependencies
+    #
+    attr_accessor :target_dependencies
+
+    # @param    [Proc] block
+    #           an optional block that configures the project through the DSL.
+    #
+    # @example  Creating a Target.
+    #
+    #           Target.new do |t|
+    #             t.name "test"
+    #           end
+    #
+    def initialize(&block)
+      block.call(self) if block_given?
+    end
+
+    # @!group Default Values
+
+    def include_files
+      @include_files ||= ["./#{@name}/**/*.*"]
+    end
+
+    def exclude_files
+      @exclude_files ||= []
+    end
+
+    def system_frameworks
+      @system_frameworks ||= default_system_frameworks_for self.platform
+    end
+
+    def target_dependencies
+      @target_dependencies ||= []
+    end
+
+    protected
+
+    def default_system_frameworks_for(platform)
+      case platform
+      when :ios
+        ['Foundation', 'UIKit']
+      when :osx
+        ['Cocoa']
+      when :tvos
+        ['Foundation', 'UIKit']
+      when :watchos
+        ['Foundation', 'UIKit', 'WatchKit']
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.<br>Use `%w` or `%W` for array of words.